kv/kvserver: TestFlowControlSendQueueRangeFeed failed

kv/kvserver.TestFlowControlSendQueueRangeFeed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20646063?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20646063?buildTab=artifacts#/) on master @ [b2be406c88674f3da35399efcf87f43fd54244b7](https://github.com/cockroachdb/cockroach/commits/b2be406c88674f3da35399efcf87f43fd54244b7):


```
=== RUN   TestFlowControlSendQueueRangeFeed
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/e13351c375d70eb2ee1cfe53170f48e2/logTestFlowControlSendQueueRangeFeed121969697
    test_log_scope.go:82: use -show-logs to present logs inline
    flow_control_integration_test.go:3412: condition failed to evaluate within 45s: from flow_control_integration_test.go:3268: expected rangefeed on n3, got n1
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e13351c375d70eb2ee1cfe53170f48e2/logTestFlowControlSendQueueRangeFeed121969697
--- FAIL: TestFlowControlSendQueueRangeFeed (45.95s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFlowControlSendQueueRangeFeed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55773

--- a/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go
@@ -623,6 +623,7 @@ func newTransportForRange(
 		return nil, err
 	}
 	replicas.OptimizeReplicaOrder(ctx, ds.st, ds.nodeIDGetter(), ds.healthFunc, ds.latencyFunc, ds.locality)
+	log.VEventf(ctx, 2, "replica order for rangefeed transport is: %s", replicas)
 	opts := SendOptions{class: defRangefeedConnClass}
 	return ds.transportFactory(opts, replicas), nil
 }

--- a/pkg/kv/kvserver/flow_control_integration_test.go
+++ b/pkg/kv/kvserver/flow_control_integration_test.go
@@ -3220,6 +3220,7 @@ func TestFlowControlSendQueueRangeSplitMerge(t *testing.T) {
 func TestFlowControlSendQueueRangeFeed(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	testutils.SetVModule(t, "dist_sender_mux_rangefeed=2,dist_sender_rangefeed=2")
 
 	// rangeFeed will create a rangefeed suitable for testing. It will start a
 	// rangefeed and return a function that can be used to stop it.

